### PR TITLE
NPT-728: Use subscribe API feature of nexus-client library in graphql

### PR DESCRIPTION
**Pull request by LokeshvarKr**
Use subscribe API feature of nexus-client library in graphql to create cache of nexus nodes and to reduce load on api-server

### CODE REVIEW

File: compiler/example/output/_rendered_templates/nexus-gql/graph/graphqlResolver.go [hunk 1]
```diff
@@ -62,6 +62,8 @@ func getRootResolver() (*model.RootRoot, error) {
             return nil, fmt.Errorf("failed to get k8s client config: %s", err)
 	    }
 	nc = nexusClient
+	nc.SubscribeAll()
+	log.Debugf("Subscribe api is called for all the nodes.")
```
Comment (xmen4xp) [inline compiler/example/output/_rendered_templates/nexus-gql/graph/graphqlResolver.go:66]:
```suggestion
	log.Debugf("Subscribed to all nodes in datamodel")
```

File: compiler/pkg/generator/template/graphql/graphqlResolver.go.tmpl [hunk 2]
```diff
@@ -65,6 +65,8 @@ func getRootResolver() (*model.{{$node.PkgName}}{{$node.NodeName}}, error) {
             return nil, fmt.Errorf("failed to get k8s client config: %s", err)
 	    }
 	nc = nexusClient
+	nc.SubscribeAll()
+	log.Debugf("Subscribe api is called for all the nodes.")
 }
```
Comment (xmen4xp) [inline compiler/pkg/generator/template/graphql/graphqlResolver.go.tmpl:70]: This brace is not indented right. Can we indent it.

File: compiler/pkg/generator/template/graphql/graphqlResolver.go.tmpl [hunk 3]
```diff
@@ -65,6 +65,8 @@ func getRootResolver() (*model.{{$node.PkgName}}{{$node.NodeName}}, error) {
             return nil, fmt.Errorf("failed to get k8s client config: %s", err)
 	    }
 	nc = nexusClient
+	nc.SubscribeAll()
```
Comment (xmen4xp) [inline compiler/pkg/generator/template/graphql/graphqlResolver.go.tmpl:68]: @LokeshvarKr You will need this in the non singleton root resolver in the else{} case below.

**Review by xmen4xp — CHANGES_REQUESTED** [review]

File: compiler/pkg/generator/template/graphql/graphqlResolver.go.tmpl (hunk 3 above)
Comment (LokeshvarKr) [reply compiler/pkg/generator/template/graphql/graphqlResolver.go.tmpl]: Thanks @xmen4xp , I have updated the code PTAL

File: compiler/pkg/generator/template/graphql/graphqlResolver.go.tmpl (hunk 2 above)
Comment (LokeshvarKr) [reply compiler/pkg/generator/template/graphql/graphqlResolver.go.tmpl]: yes, I have updated the code PTAL

File: compiler/example/output/_rendered_templates/nexus-gql/graph/graphqlResolver.go (hunk 1 above)
Comment (LokeshvarKr) [reply compiler/example/output/_rendered_templates/nexus-gql/graph/graphqlResolver.go]: updated @xmen4xp PTAL

**Review by xmen4xp — APPROVED** [review]

**Pull request merged**